### PR TITLE
Styling `Manage my account`

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/utils/AccountUtils.kt [hunk 1]
```diff
@@ -114,15 +115,32 @@ object AccountUtils : CredentialManager {
...
             withContext(Dispatchers.Main) { reloadApp?.invoke() }
+        }
+    }
+
+    private fun removeUserRealmData(context: Context, userId: Int) {
+        RealmDatabase.mailboxInfos.writeBlocking {
+            // TODO: Is it possible to directly query the ids (only the ids)?
+            val mailboxes = MailboxController.getMailboxes(userId, this)
+            mailboxes.forEach(::delete)
+        }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/AccountUtils.kt:133]: Why do you only want the IDs?

Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/utils/AccountUtils.kt]: Indeed, it makes no sens

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/res/layout/fragment_switch_user.xml [hunk 2]
```diff
@@ -15,74 +15,48 @@
...
+            style="@style/H3"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="@dimen/marginStandard"
+            android:layout_marginBottom="@dimen/marginStandard"
+            android:text="@string/buttonAddAccount"
+            android:textAllCaps="false"
+            app:icon="@drawable/ic_add"
+            app:iconGravity="start"
+            app:iconPadding="8dp"
```
Comment (sirambd) [inline app/src/main/res/layout/fragment_switch_user.xml:57]: the dimension `marginStandardSmall` exists

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 3]
```diff
@@ -59,6 +59,12 @@ object RealmDatabase {
         get() = _mailboxContent
             ?: Realm.open(RealmConfigurations.mailboxContent(AccountUtils.currentMailboxId)).also { _mailboxContent = it }
 
+    private const val appSettingsDbName = "AppSettings.realm"
+    fun userInfosDbName(userId: Int) = "User-${userId}.realm"
+    private const val mailboxInfosDbName = "MailboxInfos.realm"
+    fun mailboxContentDbNamePrefix(userId: Int) = "Mailbox-${userId}-"
+    private fun mailboxContentDbName(userId: Int, mailboxId: Int) = "${mailboxContentDbNamePrefix(userId)}${mailboxId}.realm"
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:66]: Why did you move them here?

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 4]
```diff
@@ -104,37 +110,32 @@ object RealmDatabase {
...
-        private val userInfosDbName get() = "User-${AccountUtils.currentUserId}.realm"
-        private const val mailboxInfosDbName = "MailboxInfos.realm"
-        private fun mailboxContentDbName(mailboxId: Int) = "Mailbox-${AccountUtils.currentUserId}-${mailboxId}.realm"
-
         val appSettings =
             RealmConfiguration
                 .Builder(RealmSets.appSettings)
                 .name(appSettingsDbName)
-                .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
+                // .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:117]: Put it back the way it was before, we keep them in the development phase first

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 5]
```diff
@@ -104,37 +110,32 @@ object RealmDatabase {
...
                 .build()
 
         val userInfos
             get() = RealmConfiguration
                 .Builder(RealmSets.userInfos)
-                .name(userInfosDbName)
-                .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
+                .name(userInfosDbName(AccountUtils.currentUserId))
+                // .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:124]: Put it back the way it was before, we keep them in the development phase first

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 6]
```diff
@@ -104,37 +110,32 @@ object RealmDatabase {
...
                 .build()
 
         val mailboxInfos =
             RealmConfiguration
                 .Builder(RealmSets.mailboxInfos)
                 .name(mailboxInfosDbName)
-                .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
+                // .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:131]: Put it back the way it was before, we keep them in the development phase first

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 7]
```diff
@@ -104,37 +110,32 @@ object RealmDatabase {
...
                 .build()
 
         fun mailboxContent(mailboxId: Int) =
             RealmConfiguration
                 .Builder(RealmSets.mailboxContent)
-                .name(mailboxContentDbName(mailboxId))
-                .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
+                .name(mailboxContentDbName(AccountUtils.currentUserId, mailboxId))
+                // .deleteRealmIfMigrationNeeded() // TODO: Do we want to keep this in production?
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:138]: Put it back the way it was before, we keep them in the development phase first

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/SimpleSettingView.kt [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    private val binding: ViewSimpleSettingBinding
+
+    init {
+        orientation = VERTICAL
+        binding = ViewSimpleSettingBinding.inflate(LayoutInflater.from(context), this)
+
+        if (attrs != null) {
+            val typedArray = context.obtainStyledAttributes(attrs, R.styleable.SimpleSettingView, 0, 0)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/SimpleSettingView.kt:42]: use `attrs?.getAttributes`

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/menu/SimpleSettingView.kt:42]: Fixed in #279

**Review by sirambd — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/SimpleSettingView.kt [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    private val binding: ViewSimpleSettingBinding
+
+    init {
+        orientation = VERTICAL
+        binding = ViewSimpleSettingBinding.inflate(LayoutInflater.from(context), this)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/SimpleSettingView.kt:39]: use `lazy`

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 10]
```diff
@@ -122,12 +120,34 @@ object RealmDatabase {
...
+                if (file.isDirectory) file.deleteRecursively() else file.delete()
+            }
+        }
+    }
+
     private object RealmConfig {
 
         private const val appSettingsDbName = "AppSettings.realm"
-        private val userInfoDbName get() = "User-${AccountUtils.currentUserId}.realm"
+        fun userInfoDbName(userId: Int) = "User-${userId}.realm"
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:147]: Make it private

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 11]
```diff
@@ -122,12 +120,34 @@ object RealmDatabase {
...
         private const val mailboxInfoDbName = "MailboxInfo.realm"
-        private fun mailboxContentDbName(mailboxId: Int) = "Mailbox-${AccountUtils.currentUserId}-${mailboxId}.realm"
+        fun mailboxContentDbNamePrefix(userId: Int) = "Mailbox-${userId}-"
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:149]: Make it private

**Review by sirambd — CHANGES_REQUESTED** [review]
Waiting for request changes

**Review by JorisBodin — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**